fe/estimator_abfe.py

```diff
@@ -303,10 +305,15 @@ def _deltaG(model, sys_params) -> Tuple[Tuple[float, List], np.array]:
     dG = bar_dG # use the exact answer
     dG_grad = []
 
-    # note this uses the full results, and not just sim_results
     for rhs, lhs in zip(results[-1].du_dps, results[0].du_dps):
         dG_grad.append(rhs - lhs)
 
+    if model.endpoint_correct:
+        # if we're doing endpoint correction, we need to fill in
+        # missing derivatives since zip() from above loops over
+        # the shorter array.
+        dG_grad.append(-results[0].du_dps[-1])
```

Comment:
Added more documentation on this.

Reply:
addressed in `9336d6c`

Reply:
Thanks! The added comment clarifies the current bug fix, and suggests a possibly more self-documenting data structure for a future enhancement (e.g. change the code so that any names that appear only in the comment appear instead as keys or attributes).